content/articles/2018-11-parsnip-0-0-1.Rmarkdown

```diff
@@ -111,7 +111,11 @@ car_model <- linear_reg()
 car_model
 ```
 
-That's pretty underwhelming because we haven't given it any details yet. `parsnip` offers a variety of methods to fit this general model. We will use ordinary least squares, but could also use penalized least squares too (via the lasso, ridge regression, Bayesian estimation, dropout, etc). We differentiate these cases by the ***computational engines***, which is a combination of the estimation type, such as least squares, and the _implemention_. The latter could be an R package or some other computing platform like Spark or Tensorflow. 
+That's pretty underwhelming because we haven't given it any details yet. `parsnip` offers a variety of methods to fit this general model. We will use ordinary least squares, but could also use penalized least squares too (via the lasso, ridge regression, Bayesian estimation, dropout, etc).
+
+AH: Unclear to me what it means to fit a penalized linear model via dropout. maybe clarify this is casting the problem as a neural net.
```

Comment:
Dropout is a form of regularization favored by the deep learning people. You can use it to fit a logistic or linear regression model.

Reply:
Right. This may be ignorance on my part, but for linear and logistic regression you have a convex objective with a single global optimum. Dropout doesn't change the objective, so I think that means that you're just doing coordinate descent on a random set of coordinates, and you should still converge to same the optimum unless you're doing some else like early stopping, etc. 

I've never seen someone do this and just thought it was a bit odd to suggest as a use case.

Reply:
Hmm. I'll leave it in even though it isn't a common methdology